File: OPENAPI-REST-API/swagger-client/scala-akka-http-server/src/main/scala/io/swagger/server/model/QsOrderRequest.scala
package io.swagger.server.model


/**
 * Request body to validate (putQs) or place (addQs) a QuickServer order. Server and OS template come from getNewQs.
 *
 * @param server QuickServer plan/server id from getNewQs.
 * @param password Root password for the QuickServer.
 * @param os OS template file name from getNewQs (falls back to a default if unrecognized).
 * @param tos Terms-of-service acceptance. Must be true to place the order.
 * @param comment Free-form note saved on the service row.
 */
case class QsOrderRequest (
  server: Int,
  password: String,
  os: Option[String],
  tos: Boolean,
  comment: Option[String]
)

File: OPENAPI-REST-API/openapi-client/javascript-closure-angular/API/Client/QsOrderRequest.js
goog.provide('API.Client.QsOrderRequest');

/**
 * Request body to validate (putQs) or place (addQs) a QuickServer order. Server and OS template come from getNewQs.
 * @record
 */
API.Client.QsOrderRequest = function() {}

/**
 * QuickServer plan/server id from getNewQs.
 * @type {!number}
 * @export
 */
API.Client.QsOrderRequest.prototype.server;

/**
 * Root password for the QuickServer.
 * @type {!string}
 * @export
 */
API.Client.QsOrderRequest.prototype.password;

/**
 * Terms-of-service acceptance. Must be true to place the order.
 * @type {!boolean}
 * @export
 */
API.Client.QsOrderRequest.prototype.tos;

/**
 * OS template file name from getNewQs (falls back to a default if unrecognized).
 * @type {!string}
 * @export
 */
API.Client.QsOrderRequest.prototype.os;

/**
 * Free-form note saved on the service row.
 * @type {!string}
 * @export
 */
API.Client.QsOrderRequest.prototype.comment;

